tests/ports/rp2/rp2_machine_timer.py

```diff
@@ -0,0 +1,21 @@
+from machine import Timer
+from time import sleep_ms
+
+# Test the rp2-specific adjustable tick_hz and hard/soft IRQ handlers
+# for both one-shot and periodic timers.
+
+modes = {Timer.ONE_SHOT: "one-shot", Timer.PERIODIC: "periodic"}
+kinds = {False: "soft", True: "hard"}
+
+for mode in modes:
+    for hard in kinds:
+        for hz in 1000, 2000:
+            timer = machine.Timer(
+                period=2,
+                mode=mode,
+                hard=hard,
+                tick_hz=hz,
+                callback=lambda t: print("callback", modes[mode], kinds[hard], hz),
+            )
+            sleep_ms(5)
```

Comment:
This sleep duration is a little unreliable, because sometimes the IRQ fires 4 times and other times it fires 5 times, making the test fail.

You can test this with:
```
./run-tests.py -t a0 --via-mpy -d ports/rp2
```
That's a bit more of an involved test, going via compiled .mpy files.

I suggest making the sleep not a multiple of the period, so it's more reliable to get the exact number of desired callbacks.

Reply:
Originally I had a fixed period of 2ms here which was chosen not to divide the 5ms sleep. Not sure why I switched to hz instead of 2ms, 4ms, but you're quite right, it's better with period = 2, 4.